core/embed/trezorhal/board_capabilities.c

```diff
@@ -61,9 +60,7 @@ void parse_boardloader_capabilities() {
         // not used yet, just advance pointer
         break;
       case MODEL_NAME:
-        used_length = MIN(MODEL_NAME_MAX_LENGTH, length);
-        memcpy(board_name, pos, used_length);
-        board_name[MODEL_NAME_MAX_LENGTH] = 0;
+        memcpy((uint8_t *)&board_name, pos, sizeof(uint32_t));
```

Comment:
this should first check that `length == sizeof(uint32_t)`

Reply:
fixed: https://github.com/trezor/trezor-firmware/commit/3172011eaa71ea86e25e94701f2cd9962a855891